lib/queryBuilder/graphUpserter/UpsertNode.js

```diff
@@ -96,6 +97,14 @@ function hasChanges(currentModel, upsertModel, queryProps) {
 
     if (!relations[key] && currentModel[key] !== upsertModel[key]) {
       return true;
+    } else if (relations[key] && relations[key].constructor.name === BelongsToOneRelation.name) {
+      const idColumn = currentModel.constructor.idColumn;
+      const currentModelBelongsToOneId = currentModel[key] ? currentModel[key][idColumn] : null;
+      const upsertModelBelongsToOneId = upsertModel[key] ? upsertModel[key][idColumn] : null;
+
```

Comment:
This doesn't handle the case when the relation key is not a primary key (it doesn't have to be). Also this doesn't handle composite keys or json keys. You should use the [Relation](https://github.com/Vincit/objection.js/blob/master/doc/includes/API.md#relation) class's methods (relations[key] is an instance of `Relation`). Those methods are not even documented yet since they have been added after 0.8.

You need to use the `getProp` method of `RelationProperty` class. [Here's an usage example in the code](https://github.com/Vincit/objection.js/blob/1e61de996947d5494aa5ad2cf755fb5676cf9297/lib/queryBuilder/graphInserter/BelongsToOneDependency.js#L12)

I'm sorry, I'm not nitpicking here. Objection is pretty complex beast and there are alot of corner cases that need to be supported.

Reply:
The methods and classes are documented in the `docs` folder, but not in the github.io page.

Reply:
No worries! No need to be sorry :) It's clear you're nowhere near nitpicking.